### ISSUE TITLE
sae_manifold_fit: K=2 periodic atoms fail to learn pure torus signal (R^2 ~ 0)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#174]
**gamfit 0.1.123** — given a clean 4-D torus signal embedded in R³² via a random linear mixing (noise σ=0.01), `K=2, d_atom=1, atom_topology='periodic', assignment='ibp_map'` produces a reconstruction with R² ≈ -0.001 in-sample. Two 1-D periodic atoms span the data exactly in principle, so this is well below what the configuration should achieve.

### Minimal reproducer
\`\`\`python
import numpy as np, gamfit
rng = np.random.default_rng(0); n = 600
th1 = rng.uniform(0, 2*np.pi, n); th2 = rng.uniform(0, 2*np.pi, n)
raw = np.stack([np.cos(th1), np.sin(th1), np.cos(th2), np.sin(th2)], axis=1)
Z = (raw @ rng.normal(size=(4, 32))/2.0
     + 0.01*rng.normal(size=(n, 32))).astype(np.float64)
m = gamfit.sae_manifold_fit(Z=Z, K=2, d_atom=1, atom_topology="periodic",
                            assignment="ibp_map", n_iter=40, random_state=0)
rec = m.reconstruct(Z)
ss_res = ((Z-rec)**2).sum(); ss_tot = ((Z-Z.mean(0))**2).sum()
print("R^2:", 1 - ss_res/ss_tot)   # observed ~ -0.001
\`\`\`

### Expected
R² well above 0 on a clean torus dataset (the true representation is exactly two independent S¹ atoms).

Reproducer file: /tmp/sae_topo_repro_periodic_r2.py

**Comment by SauersML** [comment]
Additional diagnostic: every entry of `model.assignments` is ~6e-37 (numerically zero), i.e. the IBP-MAP assignment matrix collapsed to no-atom-for-anyone. That fully explains the near-zero R² — the decoder produces ~zero reconstruction for every sample. So the visible bug is "no R²", but the underlying failure is "ibp_map assignment posterior went to ~0 for all (n, k)".

Same config also happens on the multicircle dataset (K=3) with both 200 and 600 samples — assignments collapse to all-zero, reconstructions collapse to mean.

**Comment by SauersML** [comment]
Broader pattern: the assignment-matrix collapse is not unique to `ibp_map`. On the same clean torus signal:

| assignment   | R²       | max(A)    | active atoms |
|--------------|----------|-----------|--------------|
| `ibp_map`    | -0.0011  | 6.006e-37 | 0            |
| `ibp`        | -0.0011  | 6.006e-37 | 0            |
| `jumprelu`   | -0.0011  | 0.0       | 0            |
| `softmax`    | crashes  | -         | -            |

So every supported gate option collapses to all-zero assignments on this clean dataset; the visible "R² ≈ 0" is just the symptom of that collapse. Same on the 3-component multicircle dataset with K=3 (`A_max = 4.9e-49`, 0 active atoms, ARI=0). Same on the K=10-over-K_true=2 overspec dataset (`A_max = 1.2e-98`, 0/10 active atoms).

(`softmax` is filed separately as #177; its failure is upstream in the Hessian-diag path.)

**Comment by SauersML** [comment]
Resolved on `main`.

Root cause: with K≥2 periodic atoms and zero-init decoder coefficients, the IBP-MAP assignment prior was the only term with a non-zero gradient at iter 0, so the joint Newton loop collapsed to A≈0 and reconstruction stayed at the mean (R² ≈ 0).

Fix: `cbedc218` added the torus SAE manifold basis and the joint LSQ decoder initialization — pyffi now seeds B_k by solving the joint normal equations on the design `[0.5·Φ_1 | 0.5·Φ_2 | …]` (mirroring the IBP-MAP soft-assignment at zero logits) with a small ridge. That gives Newton a non-degenerate starting basin.

Regression test `ibp_map_k2_periodic_torus_recovers_signal_with_lsq_init` (src/terms/sae_manifold.rs L3127-L3298, commit `554daa4a`) reproduces the exact #174 config — K=2, d_atom=1, periodic atoms, IBP-MAP, clean torus signal in R^p via a random linear mix — performs the same LSQ seed Rust-side, runs 30 arrow-Schur joint-fit steps, and asserts R² > 0.5 plus mean active-mass > 0.2 (no assignment collapse).

Closing.
